hip-service/Link/Patient/PatientVerification.cs

```diff
@@ -1,26 +1,76 @@
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Optional;
 
 namespace hip_service.Link.Patient
 {
     public class PatientVerification: IPatientVerification
     {
-        private readonly OtpVerification otpVerification;
-
-        public PatientVerification(OtpVerification otpVerification)
+        private readonly IConfiguration configuration;
+        
+        public PatientVerification(IConfiguration configuration)
         {
-            this.otpVerification = otpVerification;
+            this.configuration = configuration;
         }
-
-        public async Task<Error?> SendTokenFor(Session session)
+        
+        public async Task<OtpMessage> SendTokenFor(Session session)
         {
-            return await otpVerification.GenerateOtp(session);
+            var uri = new Uri(configuration.GetConnectionString("OTPGenerationConnection"));
+            var verifyMessage = await PostCallToOTPServer(uri
+                , CreateHttpContent(session));
+            return verifyMessage.ValueOr((OtpMessage) null);
         }
         
-        public async Task<Error?> Verify(string sessionId, string value)
+        public async Task<OtpMessage> Verify(string sessionId, string value)
         {
-            return await otpVerification.CheckOtpValue(sessionId, value);
+            var uri = new Uri(configuration.GetConnectionString("OTPVerificationConnection"));
+            var verifyOtpRequest = new OtpVerificationRequest(sessionId, value);
+            var verifyMessage = await PostCallToOTPServer(uri
+                , CreateHttpContent(verifyOtpRequest));
+            return verifyMessage.ValueOr((OtpMessage) null);
+        }
+        
+        private static HttpContent CreateHttpContent<T>(T content)  
+        {
+            var json = JsonConvert.SerializeObject(content);  
+            return new StringContent(json, Encoding.UTF8, "application/json");  
+        }
+        
+        private async Task<Option<OtpMessage>> PostCallToOTPServer (Uri serverUrl, HttpContent content)
+        {
+            var clientHandler = new HttpClientHandler
+            {
+                ServerCertificateCustomValidationCallback = (sender, cert, chain, sslPolicyErrors) => true
+            };
+            var httpClient = new HttpClient(clientHandler);
+            httpClient.DefaultRequestHeaders.Accept.Add(
+                new MediaTypeWithQualityHeaderValue("application/json"));
+            var response = await httpClient.PostAsync(serverUrl.ToString()
+                , content);
+            
+            if (response.IsSuccessStatusCode)
+            {
+                return Option.None<OtpMessage>();
+            }
+
+            if (!response.IsSuccessStatusCode)
+            {
+                var responseContent = response.Content;
+                using var reader = new StreamReader(await responseContent.ReadAsStreamAsync());
+                var result = await reader.ReadToEndAsync();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4807294)